flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java

```diff
@@ -216,6 +227,49 @@ public WebMonitorEndpoint(
 
         this.leaderElectionService = Preconditions.checkNotNull(leaderElectionService);
         this.fatalErrorHandler = Preconditions.checkNotNull(fatalErrorHandler);
+
+        this.threadInfoOperatorTracker = initializeThreadInfoTracker(executor);
+    }
+
+    private ThreadInfoOperatorTracker<OperatorThreadInfoStats> initializeThreadInfoTracker(
```

Comment:
maintenance-wise it would be good for this method to be static so we don't have to worry about accessing uninitialized fields if we re-order stuff (or rather to make this dependency more explicit).

Reply:
As discussed in another thread, I've moved the field that get's initialized by this method into `initializeHandlers`, so it has no "side-effects". Do you think static is still necessary given the changes?